deploy/charts/rook-ceph/templates/deployment.yaml

```
@@ -88,6 +88,10 @@ spec:
 {{- else }}
         - name: ROOK_HOSTPATH_REQUIRES_PRIVILEGED
           value: "{{ .Values.hostpathRequiresPrivileged }}"
+{{- end }}
+{{- if .Values.discover.customHostnameLabel }}
+        - name: ROOK_CUSTOM_HOSTNAME_LABEL
+          value: {{ .Values.discover.customHostnameLabel }}
```

`discover` refers specifically to the discover daemon settings. Since this is an overall operator setting, I'd suggest no need for a parent value name.
```suggestion
          value: {{ .Values.customHostnameLabel }}
```